contrib/mesos/pkg/scheduler/podtask/procurement.go

```diff
@@ -50,77 +62,212 @@ type AllOrNothingProcurement []Procurement
 
 // Procure runs each Procurement in the receiver list. The first Procurement func that
 // fails triggers T.Reset() and the error is returned, otherwise returns nil.
-func (a AllOrNothingProcurement) Procure(t *T, offer *mesos.Offer) error {
+func (a AllOrNothingProcurement) Procure(t *T, offer *mesos.Offer, n *api.Node, spec *Spec) error {
 	for _, p := range a {
-		if err := p(t, offer); err != nil {
-			t.Reset()
+		err := p.Procure(t, offer, n, spec)
+		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// ValidateProcurement checks that the offered resources are kosher, and if not panics.
+// NewValidateProcurement checks that the offered resources are kosher, and if not panics.
```

Comment:
This function does not procure anything, it just panics if `offer == nil`. As discussed, I'll remove this procurement.